build/lender/Lender.sol

```diff
@@ -97,8 +91,11 @@ contract Lender {
 
     // transfer from user to illuminate
     Safe.transferFrom(IErc20(u), msg.sender, address(this), a);
+
+    // Determine the fee
+    uint256 fee = a / feenominator;
     
-    uint256 returned = yield(u, y, a);
+    uint256 returned = yield(u, y, a - fee);
```

Comment:
Yeah unnecessary validation is something we err away from + in this case `fee` is derived from `a` so it cant be, and I *think* safeMath prevents that underflow now anyways?

Reply:
The [docs](https://docs.soliditylang.org/en/v0.8.0/types.html?highlight=min#division) indicate that overflows can occur for signed integers. And like you mentioned, since the fee is calculated based on the amount, it is not possible to compute a fee that is greater than the amount.